operators/src/processing/raster_kernel/kernel_function.rs

```diff
@@ -0,0 +1,208 @@
+use super::{error, RasterKernelError};
+use crate::util::{number_statistics::NumberStatistics, Result};
+use geoengine_datatypes::raster::{Grid2D, GridShape2D, GridSize, Pixel};
+use num::Integer;
+use num_traits::AsPrimitive;
+use snafu::ensure;
+
+/// A function that calculates one pixel values from a neighborhood of pixels.
+pub trait KernelFunction<P>: Sync + Send + Clone
+where
+    P: Pixel,
+    f64: AsPrimitive<P>,
+{
+    fn kernel_size(&self) -> GridShape2D;
+
+    /// Specifies the x extent beginning from the center pixel
+    fn x_radius(&self) -> usize {
+        self.kernel_size().axis_size_x() / 2
+    }
+
+    fn x_width(&self) -> usize {
+        self.kernel_size().axis_size_x()
+    }
+
+    /// Specifies the y extent beginning from the center pixel
+    fn y_width(&self) -> usize {
+        self.kernel_size().axis_size_y()
+    }
+
+    /// Specifies the x extent right of one pixel
+    fn y_radius(&self) -> usize {
+        self.kernel_size().axis_size_y() / 2
+    }
+
+    fn apply(&self, values: &[Option<P>]) -> Option<P>;
+}
+
+/// A kernel that computes the standard deviation of a neighborhood of pixels.
+#[derive(Debug, Clone, Copy)]
+pub struct StandardDeviationKernel {
+    dimensions: GridShape2D,
+}
+
+impl StandardDeviationKernel {
+    pub fn new(dimensions: GridShape2D) -> Result<Self, RasterKernelError> {
+        ensure!(
+            dimensions.axis_size_x().is_odd() && dimensions.axis_size_y().is_odd(),
+            error::DimensionsNotOdd { actual: dimensions }
+        );
+
+        Ok(Self { dimensions })
+    }
+}
+
+impl<P> KernelFunction<P> for StandardDeviationKernel
+where
+    P: Pixel,
+    f64: AsPrimitive<P>,
+{
+    fn kernel_size(&self) -> GridShape2D {
+        self.dimensions
+    }
+
+    fn apply(&self, values: &[Option<P>]) -> Option<P> {
+        if values.len() != self.dimensions.number_of_elements() {
+            return None;
+        }
+
+        let mut aggreagtor = NumberStatistics::default();
+        for value in values {
+            match value {
+                Some(v) => aggreagtor.add(*v),
+                // TODO: apply strategy here?
```

Comment:
do you mean a configuration on how to handle no data in the calculation of the stdev? Maybe clarify this comment a bit

Reply:
change the TODO in [b6b3237](https://github.com/geo-engine/geoengine/pull/644/commits/b6b323705d0d990d2ce33577c0f17192939cefcb)